util/printer/csv_printer.go

```diff
@@ -0,0 +1,124 @@
+// Copyright © 2018 Infostellar, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package printer
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"reflect"
+	"time"
+)
+
+// A Printer which output values as a CSV format.
+type CSVPrinter struct {
+	// Options of CSV format.
+	Options *CSVPrinterOptions
```

Comment:
Don't think this needs to be pointer

Reply:
I think there isn't standard rule of when to use pointer to struct. The rule I uses is when a struct is small, use struct as a value. When it is a big, I would use pointer. 
In this case, CSVPrinterOptions is small to mid size, but I thought it might have more properties in the future.

Reply:
I would also say if a struct is only used in one other location, like here where `CSVPrinterOptions` is only used by `CSVPrinter`, it's better not to use a pointer. Go will usually inline everything so there aren't many copies.